libweb3jsonrpc/Test.cpp

```diff
@@ -101,7 +102,15 @@ bool Test::test_mineBlocks(int _number)
 {
     try
     {
+        std::promise<void> allBlocksImported;
+        int importedBlocks = 0;
+        auto importHandler = m_eth.setOnBlockImport(
+            [_number, &importedBlocks, &allBlocksImported](BlockHeader const&) {
+                if (++importedBlocks == _number)
+                    allBlocksImported.set_value();
+            });
         asClientTest(m_eth).mineBlocks(_number);
+        allBlocksImported.get_future().wait();
```

Comment:
Actually it would be a good idea not to wait here indefinitely if something went wrong with the mining. 
Let's use `wait_for` here and return an error if it failed to mine in some limited amount of time (probably depending on number of blocks requested to mine, minute per block perhaps?)